### PR TITLE
Feat: add handlebars language link helper

### PR DESCRIPTION
**Pull request by falric**
Hopefully the JSDoc, README, and test will give a decent idea of what’s been added.

The screenshots are from an draft version of `node-web`, https://github.com/KTH/node-web/pull/115.

https://github.com/KTH/kth-node-web-common/assets/959319/98e3695d-e32a-4699-a08f-b27db6d601ca

https://github.com/KTH/kth-node-web-common/assets/959319/92748330-7465-4c06-a3f6-ae49886b54b3

https://github.com/KTH/kth-node-web-common/assets/959319/bd0fd5b4-6faf-4229-9706-6bd2320f6458

### CODE REVIEW

File: lib/handlebars/helpers/languageLink.test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+const handlebars = require('handlebars')
+
+const mockTranslations = {
+  label_lang_en_sv: 'English',
+  label_lang_en_en: 'Engelska',
```
Comment (nsandstrom) [inline lib/handlebars/helpers/languageLink.test.js:5]: Is _en_en and _sv_sv ever used?

File: lib/handlebars/helpers/languageLink.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  label_button_close_sv: 'Stäng',
+  label_button_close_en: 'Close',
+  label_not_translated_en: 'Den här sidan är inte översatt',
+  label_not_translated_sv: 'This page is not translated',
+  label_start_page_sv: 'Start page in English',
+  label_start_page_en: 'Startsida på svenska',
+}
+
+jest.mock('kth-node-i18n', () => ({
+  message: (key, lang) => mockTranslations[`${key}_${lang}`],
```
Comment (nsandstrom) [inline lib/handlebars/helpers/languageLink.test.js:21]: Might be nice if the format of the mock-labels where more like the ones we use in apps, ie `mockTranslations[lang][key]`

File: lib/handlebars/helpers/languageLink.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}))
+
+const { registerLanguageLinkHelper, languageLink } = require('./languageLink')
+
+jest.mock('handlebars')
+jest.mock()
```
Comment (nsandstrom) [inline lib/handlebars/helpers/languageLink.test.js:27]: Accidental empty mock?

File: README.md [hunk 4]
```diff
@@ -42,6 +42,80 @@ registerBreadcrumbHelper()
...
+```
+
+### Use handlebars helper in head
+
+Include the handlebars helper in the template.
+
+```handlebars
+<!-- Typically in server/views/partials/kthHeader.handlebars -->
+
+{{{languageLink lang}}}
```
Comment (nsandstrom) [inline README.md:105]: Is this where you "configure" the helper, passing "link", "dialogMessage" etc?

How does a more complex implementation look?

**Comment by ahrsjo** [comment]
A general question, perhaps out of scope.

Is the preferred language stored somehow for each user (localStorage, navigator.language property or something else)?

File: lib/handlebars/helpers/languageLink.test.js (hunk 1 above)
Comment (falric) [reply lib/handlebars/helpers/languageLink.test.js:5]: Nope. Not anymore. Good catch.

File: lib/handlebars/helpers/languageLink.test.js (hunk 3 above)
Comment (falric) [reply lib/handlebars/helpers/languageLink.test.js:27]: Yep. 🤦

**Comment by falric** [comment]
> A general question, perhaps out of scope.
> 
> Is the preferred language stored somehow for each user (localStorage, navigator.language property or something else)?

In a 🍪

File: lib/handlebars/helpers/languageLink.test.js (hunk 2 above)
Comment (falric) [reply lib/handlebars/helpers/languageLink.test.js:21]: Yes! Of course. Good suggestion. Much cleaner. ✨

**Review by ahrsjo — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (falric) [reply README.md:105]: Added documentation for a more complex implementation.

**Comment by nsandstrom** [comment]
Maybe not related to this particular PR, but in what pattern do we use "label" when it comes to translation? it seems like 25%  something of all keywords is prefixed "label_", but I am not sure if there is a pattern.

**Comment by falric** [comment]
> Maybe not related to this particular PR, but in what pattern do we use "label" when it comes to translation? it seems like 25% something of all keywords is prefixed "label_", but I am not sure if there is a pattern.

Ah. Yes. Naming conventions for ”message keys”. I would use _label_ to mean _”Short text that describes an element or group of elements”_. _Element_ in this case doesn’t need to be an actual DOM Element, or even the specific label element. That said, we don’t have a convention and we should have one.

**Pull request merged**